app/src/main/java/com/example/sharewhatyoucanproject/HomescreenActivity.kt

```diff
@@ -1,25 +1,52 @@
 package com.example.sharewhatyoucanproject
 
-import androidx.appcompat.app.AppCompatActivity
-import android.os.Bundle
 import android.content.Intent
+import android.os.Bundle
 import android.view.View
 import android.widget.ImageView
+import androidx.appcompat.app.AppCompatActivity
+import com.google.firebase.auth.FirebaseAuth
 
 class HomescreenActivity : AppCompatActivity() {
+    var firebaseAuth: FirebaseAuth? = null
+    var donerimg: ImageView? = null
+    var receiverimg: ImageView? = null
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_homescreen)
-        //donor action
-        val button = findViewById<View>(R.id.donerimg) as ImageView
-        button.setOnClickListener { // Do something in response to button click
-            startActivity(Intent(this@HomescreenActivity, LoginActivity::class.java))
-        }
+        firebaseAuth = FirebaseAuth.getInstance()
+        donerimg = findViewById(R.id.donerimg)
+        receiverimg = findViewById(R.id.receiverimg)
+        donerimg?.setOnClickListener(
+            View.OnClickListener {
+                val i = Intent(this@HomescreenActivity, LoginActivity::class.java)
+                i.putExtra("type", 1)
+                startActivity(i)
```

Comment:
Does "1" and "2" below have meaning?  If so, could you create self-documenting (meaningful) constants and use those instead?